### PR TITLE
Concatenate `BinaryOps` correctly for even-length `BinaryNumeralString`

### PR DESCRIPTION
**Pull request by str4d**
This fixes a bug introduced in #34. Once this is reviewed and merged, I will publish `0.6.1` and yank `0.6.0`.

### CODE REVIEW

File: src/ff1/alloc.rs [hunk 1]
```diff
@@ -57,6 +57,7 @@ impl Numeral for BigUint {
 }
 
 /// A numeral string that supports radixes in [2..2^16).
+#[cfg_attr(test, derive(Debug))]
```
Comment (nuttycom) [inline src/ff1/alloc.rs]: Neat, I hadn't seen this technique before!

File: src/ff1/proptests.rs [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+        let min_len = 0;
+        (radix, max_numeral, min_len)
+    }
+}
+
+prop_compose! {
+    fn flexible_ns()(
+        (radix, max_numeral, min_len) in valid_radix(),
+    )(
+        radix in Just(radix),
```
Comment (nuttycom) [inline src/ff1/proptests.rs:26]: Why is this line necessary? You should be able to just use `radix` directly in the body of the generator.

**Review by nuttycom — APPROVED** [review]
utACK

File: src/ff1/proptests.rs (hunk 2 above)
Comment (str4d) [reply src/ff1/proptests.rs:26]: That didn't work for me; the values in the first layer of the strategy aren't accessible beyond the second layer AFAICT.

**Review by daira — COMMENTED** [review]
Post-hoc ACK.
